pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -184,6 +210,168 @@ func (r *MachineSyncReconciler) reconcileCAPIMachinetoMAPIMachine(ctx context.Co
+			return ctrl.Result{}, utilerrors.NewAggregate([]error{conversionErr, condErr})
+		}
+
+		return ctrl.Result{}, conversionErr
+	}
+
+	for _, warning := range warns {
+		logger.Info("Warning during conversion", "warning", warning)
+		r.Recorder.Event(mapiMachine, corev1.EventTypeWarning, "ConversionWarning", warning)
+	}
+
+	newCAPIMachine.SetResourceVersion(util.GetResourceVersion(client.Object(capiMachine)))
+	newCAPIMachine.SetNamespace(r.CAPINamespace)
+	newCAPIMachine.Spec.InfrastructureRef.Namespace = r.CAPINamespace
+
+	_, infraMachine, err := r.fetchCAPIInfraResources(ctx, newCAPIMachine)
+	if err != nil && !apierrors.IsNotFound(err) {
+		fetchErr := fmt.Errorf("failed to fetch CAPI infra resources: %w", err)
+
+		if condErr := r.updateSynchronizedConditionWithPatch(
+			ctx, mapiMachine, corev1.ConditionFalse, reasonFailedToGetCAPIInfraResources, fetchErr.Error(), nil); condErr != nil {
+			return ctrl.Result{}, utilerrors.NewAggregate([]error{fetchErr, condErr})
+		}
+
+		return ctrl.Result{}, fetchErr
+	}
+
+	newCAPIInfraMachine.SetResourceVersion(util.GetResourceVersion(infraMachine))
+	newCAPIInfraMachine.SetNamespace(r.CAPINamespace)
+
+	if result, err := r.createOrUpdateCAPIInfraMachine(ctx, mapiMachine, infraMachine, newCAPIInfraMachine); err != nil {
+		return result, fmt.Errorf("unable to ensure CAPI infra machine: %w", err)
+	}
+
+	if result, err := r.createOrUpdateCAPIMachine(ctx, mapiMachine, capiMachine, newCAPIMachine); err != nil {
+		return result, fmt.Errorf("unable to ensure CAPI machine: %w", err)
+	}
+
+	return ctrl.Result{}, r.updateSynchronizedConditionWithPatch(ctx, mapiMachine, corev1.ConditionTrue,
+		consts.ReasonResourceSynchronized, messageSuccessfullySynchronized, &mapiMachine.Generation)
+}
+
+// convertMAPIToCAPIMachine converts a MAPI Machine to a CAPI Machine, selecting the correct converter based on the platform.
+func (r *MachineSyncReconciler) convertMAPIToCAPIMachine(mapiMachine *machinev1beta1.Machine) (*capiv1beta1.Machine, client.Object, []string, error) {
+	switch r.Platform {
+	case configv1.AWSPlatformType:
+		return mapi2capi.FromAWSMachineAndInfra(mapiMachine, r.Infra).ToMachineAndInfrastructureMachine() //nolint:wrapcheck
+	case configv1.PowerVSPlatformType:
+		return mapi2capi.FromPowerVSMachineAndInfra(mapiMachine, r.Infra).ToMachineAndInfrastructureMachine() //nolint:wrapcheck
+	default:
+		return nil, nil, nil, fmt.Errorf("%w: %s", errPlatformNotSupported, r.Platform)
+	}
+}
+
+// createOrUpdateCAPIInfraMachine creates a CAPI infra machine from a MAPI machine, or updates if it exists and it is out of date.
+func (r *MachineSyncReconciler) createOrUpdateCAPIInfraMachine(ctx context.Context, mapiMachine *machinev1beta1.Machine, infraMachine client.Object, newCAPIInfraMachine client.Object) (ctrl.Result, error) { //nolint:unparam
```

Comment:
What params are not being used?

Reply:
It is for the empty ctrl.Result{} that is always returned.
Stylistically I'd prefer to keep that in the function returns, as it is a method over the reconciler.